rt/frontend/jaxrs/src/main/java/org/apache/cxf/jaxrs/utils/JAXRSUtils.java

```diff
@@ -1878,4 +1891,16 @@ public static String logMessageHandlerProblem(String name, Class<?> cls, MediaTy
         return errorMessage;
     }
 
+    // copy the input stream so that it is not inadvertently closed
+    private static InputStream copyAndGetEntityStream(Message m) {
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        try {
+            IOUtils.copy(m.getContent(InputStream.class), baos);
+        } catch (IOException e) {
+            throw ExceptionUtils.toInternalServerErrorException(e, null);
+        }
+        final byte[] copiedBytes = baos.toByteArray();
+        m.setContent(InputStream.class, new ByteArrayInputStream(copiedBytes));
```

Comment:
I am wondering if we have to close the source (origin) `InputStream` here, since we replace it (dangling?) with the copy , what do you think @andymc12 ?

Reply:
Good question.  I'm running that suggestion through our automated tests and the TCK - on the surface, I think it should be fine, but just want to be sure.  The change I added looks like this:
```
InputStream origInputStream = m.getContent(InputStream.class);
        try {
            IOUtils.copy(origInputStream, baos);
        } catch (IOException e) {
            throw ExceptionUtils.toInternalServerErrorException(e, null);
        } finally {
            try {
                origInputStream.close();
            } catch (Throwable t) { /* AutoFFDC */ }
        }
```

I'm out of the office tomorrow, so I may not have the results until Monday.

Reply:
:+1: thanks a lot!

Reply:
I see Dan already merged it (thanks Dan!), but just to followup, the change had no ill effects in our automated testing - the normal functional tests and TCK all passed.

Reply:
Yep, I also rerun TCK (https://issues.apache.org/jira/browse/CXF-7996) and these failure are gone, thanks @andymc12 !